test/jdk/tools/jar/ManifestDirectoryCompression.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, Alphabet LLC. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8297875
+ * @summary jar should not compress the manifest directory entry
+ * @modules jdk.jartool
+ * @run testng ManifestDirectoryCompression
+ */
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.spi.ToolProvider;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+
+public class ManifestDirectoryCompression {
+    private static final ToolProvider JAR_TOOL =
+            ToolProvider.findFirst("jar")
+                    .orElseThrow(() -> new RuntimeException("jar tool not found"));
+
+    /** Remove dirs & files needed for test. */
+    private static void cleanup(Path dir) {
+        try {
+            if (Files.isDirectory(dir)) {
+                try (Stream<Path> s = Files.list(dir)) {
+                    s.forEach(p -> cleanup(p));
+                }
+            }
+            Files.delete(dir);
+        } catch (IOException e) {
```

Comment:
Have the method throw IOException and you do not need the catch block

Reply:
Unfortunately it's recursing on `cleanup` in the lambda, so it can't throw checked exceptions without more refactoring. This is imitating the recursive deletion approach in another jar test, I'm happy to swap this out if you'd prefer a different approach

Reply:
Thanks for the reminder regarding lambada/checked exceptions. I think this can be simplified to just write the jar and test file to the current directory and then just call

```
Files.deleteIfExists(Path.of(JAR_FILE_NAME));
Files.deleteIfExists(Path.of(FILE_NAME));
```

as they are the only files used by the test .

Reply:
Done, thanks